scripts/dcommand.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+
+import os
+import argparse
+
+
+def get_argument_parser():
+    """Creates the argument parser."""
+
+    # noinspection PyTypeChecker
+    parser = argparse.ArgumentParser(
+        description="dcommand", formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+
+    parser.add_argument(
+        "--runs_on",
+        required=True,
+        type=str,
+        help="Comma-separated IP addresses of machines",
+    )
+
+    parser.add_argument(
+        "--ssh_cmd",
+        required=False,
+        type=str,
+        default="ssh {addr}",
+        help="SSH command. Useful to utilize a pre-shared key with 'ssh -i path/to/mykey.pem ubuntu@{addr}'.",
+    )
+
+    parser.add_argument(
+        "--command",
+        required=False,
+        default="nvidia-smi | head -n 35",
+        type=str,
+        help="Command to be run through ssh onto each machine",
+    )
+
+    return parser
+
+
+def get_args():
+    """Creates the argument parser and parses any input arguments."""
+
+    parser = get_argument_parser()
+    args = parser.parse_args()
+
+    return args
+
+
+def wrap_double(text):
+    return f'"{text}"'
+
+
+def wrap_single(text):
+    return f"'{text}'"
+
+
+def wrap_single_nested(text, quote=r"'\''"):
+    return f"{quote}{text}{quote}"
+
+
+if __name__ == "__main__":
+    args = get_args()
+
+    all_addresses = args.runs_on.split(",")
+    print(f"Running on addresses {all_addresses}")
+
+    for it, addr in enumerate(all_addresses):
+        ssh_command = f"{args.ssh_cmd.format(addr=addr)} {wrap_single(args.command)}"
+
+        print(f"{it} {addr} SSH command {ssh_command}")
+        os.system(ssh_command)
+
+    print("DONE")
```

Comment:
Super optional but it might be worth letting people export environment variables that contain some of these parameters so they don't have to repeat them on every run. E.g.
```
export SSH_CMD="ssh -i path/to/mykey.pem ubuntu@{addr}"
```

Reply:
I thought about it, but I finally thought it would also induce more errors. Note that, for the `dmain.py` script, we need `ssh_cmd` to include the `-f` option to go to background, whereas for `dcommand.py` I  wanted to be blocking...

There might be a better way to do all this, so I will think about it before the final PR.

Reply:
In the meantime, I added an alternative interface such that the current set of training IP addresses can be used if no IP addresses are explicitly passed. I know it's not directly related,  but it arguable goes in the direction of decluttering the argument list...